3_extracting_features/DPutils.py

```diff
@@ -0,0 +1,220 @@
+
+    return pd.DataFrame(index_csv_data)
+
+def compile_training_locations(
+    index_csv_path: pathlib.Path,
+    segmentation_data_path: pathlib.Path,
+    save_path: pathlib.Path,
+    object: str,
+):
+    """Compile well-site-object.csv file with cell locations, saving to save_path/plate/well
+    Args:
+        index_csv_path (pathlib.Path): Path to index.csv file for object (nuc or cyto) DeepProfiler project
+        segmentation_data_path (pathlib.Path): Path to segmentation folder with .tsv locations files
+        save_path (pathlib.Path): Path to save location files
+        object (str): Object to find segmentation locations for, either "nuc" or "cyto"
+    """
+    # Reads in csv and iterate through the rows from the plate, well and site columns
+    index_csv = pd.read_csv(index_csv_path)
+    for index, row in index_csv.iterrows():
+        plate = row["Metadata_Plate"]
+        well = row["Metadata_Well"]
+        site = row["Metadata_Site"]
+
+        # Gets identifier string that matches identifier from segmented images tsvs
+        identifier_details = row["DNA"].split("/")[-1].split("_")[0:4]
+        identifier_well = identifier_details[0]
+        identifier_site = identifier_details[3]
+        identifier = f"{identifier_well}_{identifier_site}"
+
+        # Note: The name is hard coded to "Nuclei.csv" due to how the pretrained model works. Even though it should be Cytoplasm for the cyto_project,
+        # it has to be named "Nuclei" to prevent errors when running DeepProfiler (e.g saying no cells to profile when there are)
+        locations_save_path = pathlib.Path(
+            f"{save_path}/{plate}/{well}-{site}-Nuclei.csv"
+        )
+
+        # Skips a field if the locations have already been found
+        if locations_save_path.is_file():
+            print(f"{plate} + {identifier} already has locations compiled!")
+        else:
+
+            print(f"Compiling locations for {plate} + {identifier}")
+            frame_segmentations_path = pathlib.Path(
+                f"{segmentation_data_path}/{identifier}_{object}-segmented.tsv"
+            )
+
+            # Handle errors for issues like no locations file or no data within file
+            try:
+                frame_segmentations = pd.read_csv(
+                    frame_segmentations_path, delimiter="\t"
+                )
+            except:
+                print(f"No segmentation file for {frame_segmentations_path.name}")
+                continue
+            try:
+                frame_segmentations = frame_segmentations[
+                    ["Cell_ID", "Location_Center_X", "Location_Center_Y"]
+                ]
+            except KeyError:
+                print(f"No segmentation data within {frame_segmentations_path}")
+                continue
+
+            # Note: As decribed above, even though in the cyto_project it is using Cytoplasm center coords, for DeepProfiler to work with the model, it must
```

Comment:
Are we renaming these later? I hope so!

Reply:
@gwaybio From above, do you want me to create a function that the user has to use after running DP?

Reply:
I don't know - maybe! (see my response above)